src/System.Drawing.Common/tests/SystemFontsTests.cs

```diff
@@ -37,22 +37,71 @@ public void SystemFont_Get_ReturnsExpected(Func<Font> getFont)
 
         public static IEnumerable<object[]> SystemFonts_WindowsNames_TestData()
         {
-            yield return Font(() => SystemFonts.CaptionFont, "CaptionFont", "Segoe UI");
-            yield return Font(() => SystemFonts.IconTitleFont, "IconTitleFont", "Segoe UI");
-            yield return Font(() => SystemFonts.MenuFont, "MenuFont", "Segoe UI");
-            yield return Font(() => SystemFonts.MessageBoxFont, "MessageBoxFont", "Segoe UI");
-            yield return Font(() => SystemFonts.SmallCaptionFont, "SmallCaptionFont", "Segoe UI");
-            yield return Font(() => SystemFonts.StatusFont, "StatusFont", "Segoe UI");
+            int userLangId = GetUserDefaultLCID();
+            SystemFontList fonts;
 
-            bool isArabic = (GetSystemDefaultLCID() & 0x3ff) == 0x0001;
-            yield return Font(() => SystemFonts.DefaultFont, "DefaultFont", isArabic ? "Tahoma" : "Microsoft Sans Serif");
+            switch (userLangId & 0x3ff)
+            {
+                case 0x11: //ja-JP (Japanese)
+                    fonts = new SystemFontList("Yu Gothic UI");
+                    break;
+                case 0x5C: //chr-Cher-US (Cherokee)
+                    fonts = new SystemFontList("Gadugi");
+                    break;
+                case 0x12: //ko-KR (Korean)
+                    fonts = new SystemFontList("맑은 고딕");
+                    break;
+                case 0x4: //zh-TW (Traditional Chinese, Taiwan) or zh-CN (Simplified Chinese, PRC)
+                    //Although the primary language ID is the same, the fonts are different
+                    //So we have to determine by the full language ID
+                    //https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f
+                    //Assuming this doc is correct AND the font only differs by whether it's traditional or not it should work
+                    switch (userLangId & 0xFFFF)
+                    {
+                        case 0x0004: //zh-Hans
+                        case 0x7804: //zh
+                        case 0x0804: //zh-CN
+                        case 0x1004: //zh-SG
+                            fonts = new SystemFontList("Microsoft JhengHei UI");
+                            break;
+                        case 0x7C04: //zh-Hant
+                        case 0x0C04: //zh-HK
+                        case 0x1404: //zh-MO
+                        case 0x0404: //zh-TW
+                            fonts = new SystemFontList("Microsoft YaHei UI");
+                            break;
+                        default:
+                            throw new InvalidOperationException("Cannot determine system fonts");
```

Comment:
Why not default here to "Segoe UI" instead of make it fail? Is that because you're already in the Chinese culture, and according to the docs, this line should never be hit?

Reply:
From what I can see from [the latest document](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f) (See page 45), these 8 cultures seemed to be the every possibility at the moment.

If another culture with Chinese as a primary language were to be added, I think it is somewhat unlikely that they would use "Segoe UI" Because the font doesn't support any Asian characters (e.g. Chinese, Japanese, Korean).

If we were to change the fallback to something else, in my opinion it would be better to opt for either Traditional or Simplified Chinese. What's your opinion on this?

Reply:
From what I can see from [the latest document](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/70feba9f-294e-491e-b6eb-56532684c37f) (See page 45), these 8 cultures seemed to be the every possibility at the moment.

If another culture with Chinese as a primary language were to be added, I think it is somewhat unlikely that they would use "Segoe UI" Because the font doesn't support any Asian characters (e.g. Chinese, Japanese, Korean).

If we were to change the fallback to something else, in my opinion it would be better to opt for either Traditional or Simplified Chinese. What's your opinion on this?

Reply:
Now that you mention and the docs state those are the supported values, let’s throw so that if a new one is added we catch it and throw. However could you maybe include a more descriptive exception with the value we’re getting so that if this happens it is easy to fix?

Reply:
```cs
throw new InvalidOperationException("The primary language ID is Chinese, however it was not able to" + 
                                    " determine the user locale from the LCID. This indicates that there" +
                                    " could have been additions in the list of possible locales. Please" +
                                    " open a new issue with the current LCID of the user locale (Obtainable" +
                                    " via Get-WinUILanguageOverride using Powershell, or Get-WinSystemLocale" +
                                    " if the output from the cmdlet is null/empty) and the list of system fonts.");
```
Proposing this as the exception message. Is it overly verbose?

Reply:
I think it might be a little overly verbose, plus it doesn't give you the value it got. Maybe:

```cs
throw new InvalidOperationException($"The primary language ID is Chinese, however it was not able to" + 
                                    " determine the user locale from the LCID with value: {(userLangId & 0xFFFF)}.");
```

Of course with the right HEX formatting of the value.

Reply:
Ah, fair point.